services/web/server/tests/integration/01/conftest.py

```diff
@@ -0,0 +1,146 @@
+# pylint:disable=unused-argument
+
+import asyncio
+import sys
+from copy import deepcopy
+from pathlib import Path
+from typing import Callable, Dict, Tuple
+from unittest import mock
+from uuid import UUID
+
+import aiopg
+import aiopg.sa
+import pytest
+from models_library.projects import ProjectID
+from pytest_simcore.helpers.utils_login import AUserDict, log_client_in
+from servicelib.aiohttp.application import create_safe_application
+from simcore_service_webserver.application import (
+    setup_director,
+    setup_director_v2,
+    setup_exporter,
+    setup_login,
+    setup_products,
+    setup_projects,
+    setup_resource_manager,
+    setup_rest,
+    setup_security,
+    setup_session,
+    setup_socketio,
+    setup_storage,
+    setup_users,
+)
+from simcore_service_webserver.catalog import setup_catalog
+from simcore_service_webserver.db import setup_db
+from simcore_service_webserver.scicrunch.submodule_setup import (
+    setup_scicrunch_submodule,
+)
+from simcore_service_webserver.security_roles import UserRole
+from yarl import URL
+
+API_VERSION = "v0"
+API_PREFIX = f"/{API_VERSION}"
+# store only lowercase "v1", "v2", etc...
+SUPPORTED_EXPORTER_VERSIONS = {"v1", "v2"}
+
+
+@pytest.fixture
+def client(
+    loop: asyncio.AbstractEventLoop,
+    aiohttp_client: Callable,
+    app_config: Dict,
+    postgres_with_template_db: aiopg.sa.engine.Engine,
+    mock_orphaned_services: mock.Mock,
+):
+
+    cfg = deepcopy(app_config)
+
+    assert cfg["rest"]["version"] == API_VERSION
+    assert cfg["rest"]["enabled"]
+    cfg["projects"]["enabled"] = True
+    cfg["director"]["enabled"] = True
+
+    # fake config
+    app = create_safe_application(cfg)
+
+    # activates only security+restAPI sub-modules
+    setup_db(app)
+    setup_session(app)
+    setup_security(app)
+    setup_rest(app)
+    setup_login(app)
+    setup_users(app)
+    setup_socketio(app)
+    setup_projects(app)
+    setup_director(app)
+    setup_director_v2(app)
+    setup_exporter(app)
+    setup_storage(app)
+    setup_products(app)
+    setup_catalog(app)
+    setup_scicrunch_submodule(app)
+    assert setup_resource_manager(app)
+
+    yield loop.run_until_complete(
+        aiohttp_client(
+            app,
+            server_kwargs={"port": cfg["main"]["port"], "host": cfg["main"]["host"]},
+        )
+    )
+
+
+@pytest.fixture
+def login_user_and_import_study() -> Callable:
+    async def executable(client, export_version) -> Tuple[ProjectID, AUserDict]:
```

Comment:
@pcrespov : for these callables, I tend to find the yielded fct name as a minor issue (I do agree that it should not have whatever but it is private to the fixture). The function used by the test is named after the fixture, so that is the one that reallly need a good name.

- I think the callable definition is therefore important, because that is what I see in VScode and I can understand what the function expects. So in this case, when I see ```login_user_and_import_study```. from the name I do not know what I would get back without looking at the code, which is not good. 
- I think also that the fact that the userID and projectID are reversed to the name of the function is confusing. in that case I would prefer returning a NamedTuple or a dataclass where the ordering would then not be an issue
- Annotations help a lot!
- When I succeed (obviously not that often), I try to think what I get from that function, and define a simple and clear name... but of course, in the development process, suddenly some changes come in thus alleviating this.... so I try to do a second pass to rename these. and sometimes a third.

Reply:
TL:DR; I've refactored the callable to a fixture because I required it also for cleanup, the name remained the same `login_user_and_import_study`. It now has proper annotations. Please see if it feels better now.